packages/components/organisms/f-checkout/src/store/tests/analytics.module.test.js

```diff
@@ -0,0 +1,308 @@
+import Trak from '@justeat/f-trak';
+import AnalyticsModule from '../analytics.module';
+import * as mapper from '../../services/mapper';
+import { defaultState, defaultAnalyticsState } from '../../components/_tests/helpers/setup';
+
+import {
+    UPDATE_ANALYTICS_STATE,
+    UPDATE_CHANGED_FIELDS,
+    UPDATE_AUTOFILL
+} from '../mutation-types';
+
+const { actions, mutations } = AnalyticsModule;
+
+const {
+    updateState,
+    updateChangedFields,
+    trackInitialLoad,
+    trackFormInteraction
+} = actions;
+
+const checkoutState = defaultState;
+
+describe('AnalyticsModule', () => {
+    let state = AnalyticsModule.state();
+
+    it('should create default state when initialised.', () => {
+        // Assert
+        expect(state).toEqual(defaultAnalyticsState);
+    });
+
+    describe('actions ::', () => {
+        let commit;
+        let dispatch;
+        let trackEventSpy;
+        const rootState = { checkout: checkoutState };
+
+        beforeEach(() => {
+            commit = jest.fn();
+            dispatch = jest.fn();
+            state = AnalyticsModule.state();
+            trackEventSpy = jest.spyOn(Trak, 'event');
+        });
+
+        afterEach(() => {
+            jest.clearAllMocks();
+        });
+
+        describe('updateState ::', () => {
+            it(`should call ${UPDATE_ANALYTICS_STATE} with checkoutState values`, () => {
+                // Arrange
+                const expectedState = {
+                    serviceType: checkoutState.serviceType,
+                    restaurantId: checkoutState.restaurantId,
+                    basket: checkoutState.basket,
+                    isLoggedIn: checkoutState.isLoggedIn
+                };
+
+                // Act
+                updateState({ rootState, commit }, checkoutState);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_ANALYTICS_STATE, expectedState);
+            });
+
+            it(`should call ${UPDATE_AUTOFILL} if 'isLoggedIn' is true`, () => {
+                // Arrange
+                checkoutState.isLoggedIn = true;
+
+                // Act
+                updateState({ rootState, commit }, checkoutState);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_AUTOFILL, checkoutState);
+            });
+        });
+
+        describe('updateChangedFields ::', () => {
+            it(`should call ${UPDATE_CHANGED_FIELDS} with passed field`, () => {
+                // Arrange
+                const field = 'firstName';
+
+                // Act
+                updateChangedFields({ commit }, field);
```

Comment:
`updateChangedFields` expects one field. Shouldn't it be `updateChangeField` instead?

Reply:
Not sure if a typo, but did you mean `updateChangedField`?

Reply:
Changed this now